ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -974,6 +975,94 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
     return documents
 
 
+@with_traced_module
+def traced_base_tool_invoke(langchain, pin, func, instance, args, kwargs):
+    integration = langchain._datadog_integration
+    tool_input = get_argument_value(args, kwargs, 0, "input")
+    config = get_argument_value(args, kwargs, 1, "config", optional=True)
+
+    span = integration.trace(
+        pin,
+        "%s.%s.%s.%s" % (func.__module__, func.__class__.__name__, func.__name__, func.__self__.name),
+        interface_type="tool",
+    )
+
+    tool_output = None
+    try:
+        tool_attributes = ["name", "description", "metadata", "tags"]
+
+        for attribute in tool_attributes:
```

Comment:
```suggestion
        for attribute in ("name", "description", "metadata", "tags"):
```

Reply:
I'm guessing it's for memory optimization ?